packages/core/ios/Sources/DdSdk.swift

```diff
@@ -33,21 +35,27 @@ class RNDdSdk: NSObject {
         self.mainDispatchQueue = mainDispatchQueue
         super.init()
     }
-
-
+    
     @objc(initialize:withResolver:withRejecter:)
     func initialize(configuration: NSDictionary, resolve: @escaping RCTPromiseResolveBlock, reject: @escaping RCTPromiseRejectBlock) -> Void {
         // Datadog SDK init needs to happen on the main thread: https://github.com/DataDog/dd-sdk-reactnative/issues/198
         self.mainDispatchQueue.async {
+            let sdkConfiguration = configuration.asDdSdkConfiguration()
+            let frameTimeCallback = self.buildFrameTimeCallback(vitalsUpdateFrequency: self.buildVitalsUpdateFrequency(frequency: sdkConfiguration.vitalsUpdateFrequency))
+            
             if Datadog.isInitialized {
                 // Initializing the SDK twice results in Global.rum and
                 // Global.sharedTracer to be set to no-op instances
                 consolePrint("Datadog SDK is already initialized, skipping initialization.")
                 Datadog._internal._telemetry.debug(id: "datadog_react_native: RN  SDK was already initialized in native", message: "RN SDK was already initialized in native")
+                
+                // This block is called when SDK is reinitialized and the javascript has been wiped out.
+                // In this case, we need to restart the refresh rate monitor, as the javascript thread 
+                // appears to change at that moment.
+                JsRefreshRate.init().start(frameTimeCallback)
```

Comment:
After applying my suggestion the objc implementation, you can do:
```suggestion
                JsRefreshRate.shared.start(frameTimeCallback)
```